File: files/CSEC 29.12.R
library(tidyverse)
library(readxl)

path = "files/Excel Challenge 29th Dec.xlsx"
input = read_excel(path, range = "B2:B9")
test  = read_excel(path, range = "D2:D15")

result = input %>%
  separate_rows(Problem, sep = "") %>%
  filter(Problem != "")

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: files/2024-12-29/CSEC 29.12.py
import pandas as pd

path = "files/Excel Challenge 29th Dec.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=8)
test = pd.read_excel(path, usecols="D", skiprows=1, nrows=14)

input = input.dropna().assign(Problem=input['Problem'].str.split('')).explode('Problem').query('Problem != ""').reset_index(drop=True)

print(input['Problem'].equals(test['Solution Required'])) # True